---
title: "A1_seq_analysis"
author: "Maximiliana Bogan"
date: "2/20/2020"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

## install packages
```{r}
#install.packages('knitr')
#source("http://bioconductor.org/biocLite.R")
#biocLite("phyloseq") #note if they ask to update packages selece "a" option
#install.packages('ggplot2')
#install.packages('vegan')
#install.packages("dplyr")
#biocLite("dada2")
#install.packages('corrplot')
#install.packages('tidyverse')
#install.packages("seqinr")
#install.packages('gplots')
```

## load libraries
```{r}
require('phyloseq')
require('ggplot2')
require('vegan')
library("dplyr")
library(dada2); packageVersion("dada2")
library(tidyr)
library(dplyr)
library(seqinr)
library(gplots)
library("RColorBrewer")
```

## execute heatmap
```{r}
#read in sequences
seqs <- read.alignment("~/200331_9_species_alignment_MB.fasta", format = "fasta")

#change names of sequences
seqs$nam <- c('A. femoralis','D. tinctorius','E. tricolor', 'H. fleishmanni','M. aurantiaca','N. parkeri','O. sylvatica','R. catesbeiana', 'R. imitator'  )

#create distance matrix w/ pairwise comparisons of sequence identity
aa_data <- dist.alignment(seqs, matrix = "similarity")
#convert to matrix
aa_identity <- as.matrix(aa_data)

#create percentages from similarity matrix for more intuitive display
aa_identity_percentages <- round((1-aa_identity)*100, digits = 2)

#check values
aa_identity         
range(aa_identity)   #ensure that similarity is not greater than 1

#############TO SAVE IMAGE#############
#png("200412_9_spec_heatmap", width = 11, height = 8, res = 600, units = "in", )
#######################################

aa_identity_heatmap <- 
  heatmap.2(aa_identity,
                              #define label size
                                 cexRow = 2, 
                                 cexCol = 2, 
                              #set title
                                 main = "Saxiphilin Amino Acid Sequence Identity (%)", 
                              #no dendrogram
                                 dendrogram = 'none',
                              #insert % identity values to cells
                                 cellnote = aa_identity_percentages,
                                 notecol = "black",
                                 notecex = 2,
                              #set color of cells and remove weird trace thing
                                 col = rev(brewer.pal(n = 9, name = 'YlOrRd')), 
                                 trace = "none", 
                                 tracecol = NA, 
                              #adjust label orientation and make column labels diagonal
                                 adjRow=c(0, 1), 
                                 srtCol=45, 
                                 adjCol=c(.9,1),
                              #set layout matrix
                                 lmat=rbind( c(0, 3), c(2,1), c(0,4) ), lhei=c(1.5, 4, 2),
                              #no key
                                 key = FALSE,
                              #set margins
                                 margins = c(0,13)
            
                                 )

#############TO SAVE IMAGE#############
#dev.off()
#######################################
```
